app/assets/javascripts/helpers/ResizerPlugin.es6

```diff
@@ -0,0 +1,100 @@
+// @license magnet:?xt=urn:btih:0b31508aeb0634b347b8270c7bee4d411b5d4109&dn=agpl-3.0.txt AGPL-v3-or-Later
+/**
+ * Custom resizer plugin for Uppy. Resizes image files if necessary using Pica image processing library.
+ */
+class Resizer extends Uppy.Core.Plugin {
+  constructor(uppy, options) {
+    super(uppy, options);
+    this.id = options.id || 'Resizer';
+    this.type = 'modifier';
+    this.maxSize = options.maxSize;
+    this.maxFileSize = options.maxFileSize;
+
+    this.prepareUpload = this.prepareUpload.bind(this);
+    this.resize = this.resize.bind(this);
+    this.pica = window.pica();
+  }
+
+  /**
+   * Resizes the file if necessary based on conditions.
+   * If the file exceeds the file size limit it will either be resized to maximum dimensions or only recompressed as JPEG (e.g. for huge PNG files not exceeding the dimension limit)
+   * @param file The uppy file
+   */
+  resize(file) {
+    return new Promise((resolve, reject) => {
+      const img = new Image();
+      img.src = URL.createObjectURL(file.data);
+      const canvas = document.createElement('canvas');
+
+      img.onload = () => {
+        if (img.width > this.maxSize || img.height > this.maxSize) {
+          this.uppy.log(`[Resize] ${file.name}: Image dimensions ${img.width}x${img.height} are greater than max size ${this.maxSize}. Resizing image...`);
+          let ratio;
+
+          if (img.height > img.width) {
+            ratio = img.width / img.height;
+            canvas.width = this.maxSize * ratio;
+            canvas.height = this.maxSize;
+          } else {
+            ratio = img.height / img.width;
+            canvas.width = this.maxSize;
+            canvas.height = this.maxSize * ratio;
+          }
+          this.uppy.log(`[Resize] ${file.name}: New image dimensions: ${canvas.width}x${canvas.height}...`);
+        } else {
+          this.uppy.log(`[Resize] ${file.name}: Image dimensions ${img.width}x${img.height} are smaller than max size ${this.maxSize}. Saving to JPEG without resizing...`);
+          canvas.width = img.width;
+          canvas.height = img.height;
+        }
+
+        this.pica.resize(img, canvas)
+          .then(result => this.pica.toBlob(result, 'image/jpeg', 0.85))
+          .then(blob => resolve(blob));
+      }
+    });
+  }
+
+  prepareUpload(fileIDs) {
+    const promises = fileIDs.map((fileID) => {
+      const file = this.uppy.getFile(fileID);
+      if (file.type.split('/')[0] !== 'image' || (file.size <= this.maxFileSize)) {
+        return;
+      }
+      this.uppy.log(`[Resize] ${file.name}: File size ${file.size} exceeds max file size of ${this.maxFileSize}.`);
+
+      return this.resize(file).then((blob) => {
+        // since size and eventually type are changing, we have to overwrite several redundant information in the file object
+        const filename = file.name.replace(/\.png$/i, '.jpg')
+        const processedFile = {
+          ...file,
```

Comment:
[Yo, listen up here's a story, about a contributor who tried to use new features](https://www.youtube.com/watch?v=68ugkg9RePc)...

`pronto` will complain about this, and it will complain about this very hard. In fact, it will claim this is a syntax error, although we both know the object spread operator is production-ready and shipped in all major browsers except for Edge (which I, quite frankly, no longer care about, as it works in Edgeium).

The reason `pronto` will yell at you is because `pronto` is not using the `eslint` out of your path. Instead, it calls to `pronto-eslint`, which in turns calls to `eslintrb`, [which hasn't been updated since 2016](https://github.com/zendesk/eslintrb), which still is using`eslint` 2.7.0. Obviously, object spreads are quite foreign to a eslint parser coming straight out of the early Mesolithic Age.

There is nothing I can do to make `pronto` happy with this, besides forking `eslintrb` and forking `pronto-eslint`, which is not going to happen to preserve the minds of everyone working on this project.

I hate writing this suggestion as much as you will hate it when reading, but can you rewrite that using `Object.assign({}, file, { ... })`, please?

We'll be in a better place, one day. I promise.